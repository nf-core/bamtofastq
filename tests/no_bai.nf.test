nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    tag "pipeline_bamtofastq"
    tag "cpu"

    test("-profile test --input test_bam_samplesheet_no_bai.csv") {

        when {
            params {
                outdir = "$outputDir"
                // Input data
                input = "https://raw.githubusercontent.com/nf-core/test-datasets/bamtofastq/samplesheet/test_bam_samplesheet_no_bai.csv"
                fasta = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/genome.fasta'

                // Genome references
                genome = null
                igenomes_ignore = true
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // bam_files: All bam files
            def bam_files  = getAllFilesFromDir(params.outdir, include: ['**/*.bam'])
            // cram_files: All cram files
            def cram_files  = getAllFilesFromDir(params.outdir, include: ['**/*.cram'])
            // Fasta file for cram verification with nft-bam
            def fasta_base  = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
            def fasta       = fasta_base + 'genomics/homo_sapiens/genome/genome.fasta'
            assert workflow.success
            assertAll(
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_bamtofastq_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name.isEmpty() ? 'No stable paths' : stable_name,
                    // All files with stable contents
                    stable_path.isEmpty() ? 'No stable content' : stable_path,
                    // All cram files
                    bam_files.isEmpty() ? 'No BAM files' : bam_files.collect { file -> file.getName() + ":md5," +  bam(file.toString()).readsMD5 },
                    // All cram files
                    cram_files.isEmpty() ? 'No CRAM files' : cram_files.collect { file -> file.getName() + ":md5," +  cram(file.toString(), fasta).readsMD5 },
                ).match() }
            )
        }
    }
}
