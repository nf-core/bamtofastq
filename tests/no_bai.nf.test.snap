{
    "-profile test --input test_bam_samplesheet_no_bai.csv": {
        "content": [
            40,
            {
                "CAT_FASTQ": {
                    "cat": 9.5
                },
                "CHECK_IF_PAIRED_END": {
                    "samtools": 1.21
                },
                "COLLATE_FASTQ_MAP": {
                    "samtools": 1.21
                },
                "COLLATE_FASTQ_UNMAP": {
                    "samtools": 1.21
                },
                "FASTQC_POST_CONVERSION": {
                    "fastqc": "0.12.1"
                },
                "FASTQC_PRE_CONVERSION": {
                    "fastqc": "0.12.1"
                },
                "SAMTOOLS_COLLATEFASTQ_SINGLE_END": {
                    "samtools": 1.21
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "SAMTOOLS_FLAGSTAT": {
                    "samtools": 1.21
                },
                "SAMTOOLS_IDXSTATS": {
                    "samtools": 1.21
                },
                "SAMTOOLS_INDEX": {
                    "samtools": 1.21
                },
                "SAMTOOLS_MERGE_UNMAP": {
                    "samtools": 1.21
                },
                "SAMTOOLS_STATS": {
                    "samtools": 1.21
                },
                "SAMTOOLS_VIEW_MAP_MAP": {
                    "samtools": 1.21
                },
                "SAMTOOLS_VIEW_MAP_UNMAP": {
                    "samtools": 1.21
                },
                "SAMTOOLS_VIEW_UNMAP_MAP": {
                    "samtools": 1.21
                },
                "SAMTOOLS_VIEW_UNMAP_UNMAP": {
                    "samtools": 1.21
                },
                "Workflow": {
                    "nf-core/bamtofastq": "v2.1.2dev"
                }
            },
            [
                "fastqc",
                "fastqc/test.post_conversion_1_fastqc.html",
                "fastqc/test.post_conversion_1_fastqc.zip",
                "fastqc/test.post_conversion_2_fastqc.html",
                "fastqc/test.post_conversion_2_fastqc.zip",
                "fastqc/test.pre_conversion_fastqc.html",
                "fastqc/test.pre_conversion_fastqc.zip",
                "fastqc/test2.post_conversion_1_fastqc.html",
                "fastqc/test2.post_conversion_1_fastqc.zip",
                "fastqc/test2.post_conversion_2_fastqc.html",
                "fastqc/test2.post_conversion_2_fastqc.zip",
                "fastqc/test2.pre_conversion_fastqc.html",
                "fastqc/test2.pre_conversion_fastqc.zip",
                "fastqc/test3.post_conversion_fastqc.html",
                "fastqc/test3.post_conversion_fastqc.zip",
                "fastqc/test3.pre_conversion_fastqc.html",
                "fastqc/test3.pre_conversion_fastqc.zip",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_bamtofastq_software_mqc_versions.yml",
                "reads",
                "reads/test2_1.merged.fastq.gz",
                "reads/test2_2.merged.fastq.gz",
                "reads/test3_other.fq.gz",
                "reads/test_1.merged.fastq.gz",
                "reads/test_2.merged.fastq.gz",
                "samtools",
                "samtools/test.flagstat",
                "samtools/test.idxstats",
                "samtools/test.stats",
                "samtools/test2.flagstat",
                "samtools/test2.idxstats",
                "samtools/test2.stats",
                "samtools/test3.flagstat",
                "samtools/test3.idxstats",
                "samtools/test3.stats"
            ],
            [
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
                "test2_1.merged.fastq.gz:md5,a4984fb5a60f80ccbfbfcf1cef9daaf3",
                "test2_2.merged.fastq.gz:md5,bde23801e2d60604afb3bd0b0ab014ab",
                "test3_other.fq.gz:md5,d8a935709a2519d0af2b0e28631962f7",
                "test_1.merged.fastq.gz:md5,9b0c08862a5e47e842a485ddad31b871",
                "test_2.merged.fastq.gz:md5,40e17982d1231e561d6420234ae29357",
                "test.flagstat:md5,a53f3d26e2e9851f7d528442bbfe9781",
                "test.idxstats:md5,e179601fa7b8ebce81ac3765206f6c15",
                "test.stats:md5,a15b3a5e59337db312d66020c7bb93ac",
                "test2.flagstat:md5,acbd95030a362a7e6491faa87d1b9c2c",
                "test2.idxstats:md5,90f01313bc89c7fd096aad051cd71b66",
                "test2.stats:md5,fae549c0c944f6cc34dd6d9b9569e3d3",
                "test3.flagstat:md5,7b43b975b0a395040bf1ee3c48f058e7",
                "test3.idxstats:md5,094096b6f303cc18892e1e027c87f273",
                "test3.stats:md5,a2994534537147cf4f3185bd867bc220"
            ],
            "No BAM files",
            "No CRAM files"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-06T09:39:04.205418"
    }
}
